File: app/src/main/java/com/example/smileapp/models/Appointment.java
package com.example.smileapp.models;

import androidx.room.Entity;
import androidx.room.PrimaryKey;
import java.util.Date;

@Entity(tableName = "appointments")
public class Appointment {
    @PrimaryKey(autoGenerate = true)
    public int id;
    public String childId;
    public String childName;
    public String doctorId;
    public long date; // Storing as timestamp for simplicity in Room
    public String type;
    public String status; // "upcoming", "completed"

    public Appointment() {}

    public Appointment(String childId, String childName, String doctorId, long date, String type) {
        this.childId = childId;
        this.childName = childName;
        this.doctorId = doctorId;
        this.date = date;
        this.type = type;
        this.status = "upcoming";
    }
}

File: app/src/main/java/com/example/smileapp/models/Appointment.kt
package com.example.smileapp.models

import androidx.room.Entity
import androidx.room.PrimaryKey
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
@Entity(tableName = "appointments")
data class Appointment(
    @PrimaryKey(autoGenerate = true)
    @JvmField var id: Int = 0,
    
    @SerialName("child_id")
    @JvmField var childId: String = "",
    
    @SerialName("child_name")
    @JvmField var childName: String = "",
    
    @SerialName("doctor_id")
    @JvmField var doctorId: String = "",
    
    @SerialName("appt_date")
    @JvmField var date: Long = 0,
    
    @JvmField var type: String = "",
    @JvmField var status: String = "upcoming"
) {
    // Constructor for Java compatibility
    constructor(childId: String, childName: String, doctorId: String, date: Long, type: String) : this(
        id = 0,
        childId = childId,
        childName = childName,
        doctorId = doctorId,
        date = date,
        type = type,
        status = "upcoming"
    )
}
